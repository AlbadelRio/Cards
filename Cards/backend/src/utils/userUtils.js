function dataChanges(body, userData, userFound) {
  switch (body.type) {
    case 'cardsAdquired':
      userData.cardsAdquired.push(body.cardId);
      userFound.save();
      break;

    case 'difficultCards':
      userData.difficultCards.push(body.cardId);
      userFound.save();
      break;
    default:
      break;
  }
}

module.exports = { dataChanges };
