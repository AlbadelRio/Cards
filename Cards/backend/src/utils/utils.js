function DataChanges(body, userData, userFound) {
  switch (body.type) {
    case 'cardsAdquired':
      userData.cardsAdquired.push(body.cardId);
      userFound.save();
      break;

    case 'difficultCards':
      userData.difficultCards.push(body.cardId);
      userFound.save();
      break;
    case 'stats': {
      const existingDate = userData.stats.find((data) => data.date === body.date);
      if (!existingDate) {
        userData.stats.push(body);
        userFound.save();
      } else {
        existingDate.stats.update(
          {
            $push: { timePlayed: body.time, percentages: body.percentage },
            $sort: { score: -1 }
          }

        );
      } }
      break;
    default:
      break;
  }
}

module.exports = { DataChanges };
