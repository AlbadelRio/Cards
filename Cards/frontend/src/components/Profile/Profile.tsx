import React, { useState } from 'react';
import {
  View, TextInput, TouchableOpacity, Text, Pressable
} from 'react-native';
import { useDispatch } from 'react-redux';
import LinearGradient from 'react-native-linear-gradient';
import { login } from '../../redux/actions/authActionCreators';

import styles from './profileStyles';

export default function Profile({ navigation }:any) {
  const dispatch = useDispatch();
  function handleLogin(emailText:any, passwordText:any) {
    if (emailText) {
      dispatch(login({
        email: emailText,
        password: passwordText
      }));
    }
  }

  const newAccountHandler = () => { navigation.navigate('CreateAccount'); };
  const [emailText, setEmail] = useState('');
  const [passwordText, setPassword] = useState('');
  return (
    <LinearGradient
      start={{ x: 0, y: 1 }}
      end={{ x: 0, y: 0 }}
      colors={['#222222', '#EAECF2']}
      style={styles.linearGradient}
    >

      <View>
        <Text
          style={styles.title}
        >
          CARDS

        </Text>
      </View>
      <View style={styles.container}>
        <TextInput
          testID="emailInput"
          style={styles.email}
          value={emailText}
          onChangeText={(text:any) => setEmail(text)}
          placeholder="email"
        />

        <TextInput
          testID="passwordInput"
          style={styles.email}
          value={passwordText}
          onChangeText={(text:any) => setPassword(text)}
          placeholder="password"
        />
        <TouchableOpacity
          style={styles.button}
          onPress={() => { handleLogin(emailText, passwordText); }}
        >
          <Text
            style={styles.text}
          >
            GET STARTED

          </Text>
        </TouchableOpacity>
      </View>

      <Pressable
        onPress={newAccountHandler}
      >
        <Text
          style={styles.pressableText}
        >
          NEW ACCOUNT

        </Text>
      </Pressable>
    </LinearGradient>
  );
}
