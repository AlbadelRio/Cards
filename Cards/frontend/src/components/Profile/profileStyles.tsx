import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  component: {
    height: '100%',
    backgroundColor: '#222222'
  },
  title: {
    fontSize: 60,
    color: '#E5E5E5',
    textAlign: 'center',
    marginTop: 180,
    fontWeight: 'bold'
  },
  container: {
    marginTop: 85,
    height: 300,
    alignSelf: 'center'
  },
  email: {
    width: 300,
    height: 50,
    padding: 30,
    margin: 5,
    backgroundColor: '#DCDCDC',
    flex: 1,
    alignSelf: 'center',
    borderRadius: 5

  },
  button: {
    width: 300,
    padding: 25,
    height: 70,
    marginTop: 50,
    backgroundColor: '#5EBBB0',
    flex: 1,
    alignSelf: 'center',
    borderRadius: 5
  },
  text: {
    fontSize: 18,
    color: '#222222',
    textAlign: 'center'
  },
  pressableText: {
    fontSize: 14,
    color: '#E5E5E5',
    marginTop: 60,
    marginBottom: 20,
    marginRight: 30,
    textAlign: 'right'
  }
});
