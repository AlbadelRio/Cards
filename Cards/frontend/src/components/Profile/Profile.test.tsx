import React from 'react';
import { render, fireEvent } from '../../utils/test.utils';

import Profile from './Profile';

const navigation = { navigte: jest.fn() };
jest.mock('../../redux/actions/authActionCreator', () => ({
  login: jest.fn()
}));

describe('Given a Cards component', () => {
  let screen:any;
  describe('when it is rendered', () => {
    beforeEach(() => {
      screen = render(<Profile navigation={navigation} />);
    });
    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });
  });

  describe('And email input text is changed', () => {
    test('Then the text typed should be renderer', () => {
      const email = screen.getByPlaceholderText('emailInput');
      const setEmail = jest.fn();
      fireEvent(email, 'onChangeText', setEmail);
      fireEvent.changeText(email, 'onChangeText');
      expect(screen.queryByDisplayValue(/email@email/i)).not.toBe(null);
    });
  });
  describe('And email input text is changed', () => {
    test('Then the text typed should be renderer', () => {
      const password = screen.getByPlaceholderText('passwordInput');
      const setPassword = jest.fn();
      fireEvent(password, 'onChangeText', setPassword);
      fireEvent.changeText(password, 'onChangeText');
      expect(screen.queryByDisplayValue(/email@email/i)).not.toBe(null);
    });
  });
});
