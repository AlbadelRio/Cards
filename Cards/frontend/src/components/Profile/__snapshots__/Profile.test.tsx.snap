// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Cards component when it is rendered Then should match the snapshot 1`] = `
<BVLinearGradient
  colors={
    Array [
      4280427042,
      4293586162,
    ]
  }
  endPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0,
      "y": 1,
    }
  }
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#E5E5E5",
          "fontSize": 60,
          "fontWeight": "bold",
          "marginTop": 180,
          "textAlign": "center",
        }
      }
    >
      CARDS
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "height": 300,
        "marginTop": 85,
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="email"
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#DCDCDC",
          "borderRadius": 5,
          "flex": 1,
          "height": 50,
          "margin": 5,
          "padding": 30,
          "width": 300,
        }
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="password"
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#DCDCDC",
          "borderRadius": 5,
          "flex": 1,
          "height": 50,
          "margin": 5,
          "padding": 30,
          "width": 300,
        }
      }
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "backgroundColor": "#5EBBB0",
          "borderRadius": 5,
          "flex": 1,
          "height": 70,
          "marginTop": 50,
          "opacity": 1,
          "padding": 25,
          "width": 300,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#222222",
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        GET STARTED
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Object {
          "color": "#E5E5E5",
          "fontSize": 14,
          "marginBottom": 20,
          "marginRight": 30,
          "marginTop": 60,
          "textAlign": "right",
        }
      }
    >
      NEW ACCOUNT
    </Text>
  </View>
</BVLinearGradient>
`;
